### PR TITLE
fix macos portability issues (third batch)

### PR DESCRIPTION
**Pull request by garlick**
This fixes some more macos portability issues.

### CODE REVIEW

File: src/common/libutil/basename.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/* This is what glibc basename(3) does more or less.
+ * https://github.com/lattera/glibc/blob/master/string/basename.c
+ */
+
+char *basename_simple (const char *path)
+{
+    char *p = strrchr (path, '/');
+    return p ? p + 1 : (char *)path;
+}
```
Comment (grondo) [inline src/common/libutil/basename.c:26]: Nice. Does this undo the need for 92c2ece8206d1b2940de54685b64120754efcf2b if all uses of basename were updated to `basename_simple()`?

I know these are just portability updates, but perhaps we can update the code in `shell/doom.c` that does the same thing to use `basename_simple.c` instead:

https://github.com/flux-framework/flux-core/blob/11ad605648fafbaf4837ceda2a7751a5d53efb3e/src/shell/doom.c#L100-L106

**Review by grondo — APPROVED** [review]
LGTM!

File: src/common/libutil/basename.c (hunk 1 above)
Comment (garlick) [reply src/common/libutil/basename.c:26]: Oh sure, we can do that.  I was trying to have a light touch but this makes sense.

**Comment by garlick** [comment]
Thanks!  Everything is using `basename_simple()` including the local function you pointed out and one more I found.  I'll set MWP.

**Pull request merged**